### PR TITLE
share-preview: init at 0.3.0

### PR DESCRIPTION
**Pull request by raboof**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by gaykitty** [comment]
Result of `nixpkgs-review pr 256453` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>share-preview</li>
  </ul>
</details>

**Comment by raboof** [comment]
On second thought this seems very niche, so a flake outside of nixpkgs might make more sense? WDYT?

**Comment by gaykitty** [comment]
Personally I want to see all of the Gnome Circle apps, including this one in nixpkgs.

**Comment by raboof** [comment]
OK, I'm not against it, feel free to add yourself as maintainer because I might drop it if it becomes cumbersome :)

**Comment by gaykitty** [comment]
You can add me to the list in the nix file if you want.

**Comment by raboof** [comment]
> You can add me to the list in the nix file if you want.

updated!

File: pkgs/by-name/sh/share-preview/package.nix [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+  meta = with lib; {
+    description = "Test social media cards locally";
+    longDescription = ''
+      Preview and debug websites metadata tags for social media share
+    '';
+    homepage = "https://github.com/rafaelmardojai/share-preview";
+    license = licenses.gpl3Only;
+    maintainers = with maintainers; [ gaykitty ];
+    platforms = platforms.all;
```
Comment (Aleksanaa) [inline pkgs/by-name/sh/share-preview/package.nix:68]: `mainProgram` is always needed now, despite being the same with `pname`.

**Pull request closed without merging**